provider/pkg/provider/provider.go

```diff
@@ -1386,6 +1385,14 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 			}
 		}
 	}
+	if metadata.IsGenerateName(newInputs, news) {
+		if k.serverSideApplyMode {
+			return nil, fmt.Errorf("the `.metadata.generateName` field is not supported in Server-Side Apply mode")
+		}
+		if k.yamlRenderMode {
+			return nil, fmt.Errorf("the `.metadata.generateName` field is not supported in YAML rendering mode")
+		}
+	}
```

Comment:
Would it be better to ignore the generateName field in these cases rather than erroring? You could still print a warning, but that would be less disruptive.

Reply:
Error-ing would indicate an issue with the user's code/manifest and that they should fix it - similar to conflicts with SSA. I'm on the stance of returning an error for the user to resolve, as ignoring this field will result in the name used for the object being different to the intent of the declared manifest.